### PR TITLE
Docker image for Chocolatey

### LINKED ISSUE #1709: Create a chocolatey docker image
# Task Description

We need to create a chocolatey docker image so we can test `choco` in `strongbox-web-integration-tests`.

# Task Relationships

This task:
* Relates to: #1685, #1662, #1113

# Useful Links

* [Docker multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/)
* [Dockerfile.alpine.jdk(8|11)-mvn3.6-mono5-nuget3.4](https://github.com/strongbox/ci-build-images/blob/master/images/alpine/jdk8/Dockerfile.alpine.jdk8-mvn3.6-mono5-nuget3.4)
* [Example of choco image](https://hub.docker.com/r/thinkco/chocolatey/dockerfile)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @steve-todorov

### PR DESCRIPTION
**Pull request by anki2189**
# Pull Request Description

This pull request closes https://github.com/strongbox/strongbox/issues/1709

# Acceptance Test

* [X] Running `./build.sh` successfully builds all images.

# Questions

* Does this pull request somehow break backward compatibility? 
  * [ ] Yes, it will require fixing build jobs.
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: common-scripts/setup-choco.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	#!/bin/bash \n
+	set -e \n
+	function cleanup { \n
+	    if [ $PWD != "/" ] && [ -d opt ]; then \n
+	        rm -rf opt \n
+	    fi \n
+	} \n
+	trap cleanup EXIT \n
+	mono /opt/chocolatey/choco.exe "$@" --allow-unofficial \n
+	'  > /usr/bin/choco
```
Comment (steve-todorov) [inline common-scripts/setup-choco.sh]: This should be aligned to the beginning (so it looks "normal" when you open it from the container)

```suggestion
	echo $'\n
#!/bin/bash \n
set -e \n
function cleanup { \n
    if [ $PWD != "/" ] && [ -d opt ]; then \n
        rm -rf opt \n
    fi \n
} \n
trap cleanup EXIT \n
mono /opt/chocolatey/choco.exe "$@" --allow-unofficial \n
'  > /usr/bin/choco 
```

Comment (anki2189) [reply common-scripts/setup-choco.sh]: Fixed.

Comment (steve-todorov) [reply common-scripts/setup-choco.sh]: Thanks!

File: images/alpine/jdk8/Dockerfile.alpine.jdk8-mvn3.6-mono5-nuget3.4-choco0.10 [hunk 2]
```diff
@@ -0,0 +1,22 @@
+FROM mono:3.12.1 as builder
+
+ENV CHOCO_VERSION=master
```
Comment (steve-todorov) [inline images/alpine/jdk8/Dockerfile.alpine.jdk8-mvn3.6-mono5-nuget3.4-choco0.10]: Sorry, I mislead you a bit in our discussions. Could you please switch the version to `9bb4246354188e4205b5dbcae6f8638351eccb13`? We should keep a `fixed revision` until they release `0.10.16`. I don't know if they are planning on introducing breaking changes and it looks like the current `master` revision works for us.

```suggestion
ENV CHOCO_VERSION=9bb4246354188e4205b5dbcae6f8638351eccb13
```

Comment (anki2189) [reply images/alpine/jdk8/Dockerfile.alpine.jdk8-mvn3.6-mono5-nuget3.4-choco0.10]: Changed.

**Review by steve-todorov — APPROVED** [review]
LGTM! Many thanks for making this happen! :)

**Comment by carlspring** [comment]
Thanks for your help, @anki2189 ! :)

**Pull request merged**